lib/ui/common/InputAlert.dart

```diff
@@ -0,0 +1,222 @@
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      type: MaterialType.transparency,
+      child: Center(
+        child: Container(
+          margin: _Constants.alertEdgePadding,
+          decoration: BoxDecoration(
+            borderRadius: _Constants.cornerRadius,
+            color: _style.backgroundColor,
+          ),
+          child: Padding(
+            padding: _Constants.alertInnerPadding,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                Text(
+                  _viewModel.titleText,
+                  style: _style.titleTextStyle,
+                ),
+                SizedBox(
+                  height: _Constants.titleLineSpace,
+                ),
+                Column(
+                  mainAxisSize: MainAxisSize.min,
+                  children: <Widget>[
+                    TextField(
+                      controller: _textFieldController,
+                      decoration:
+                          InputDecoration(hintText: _viewModel.hint),
+                    ),
+                    SizedBox(
+                      height: _Constants.detailLineSpace,
+                    ),
+                    Row(
+                      children: _buildAction(context),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  List<Widget> _buildAction(BuildContext context) {
+    List<Widget> listBuilder = [
+      Expanded(
+        child: RoundedButton(
+          viewModel: RoundedButtonViewModel(
+            title: _viewModel.cancelActionText,
+            onTap: () => Navigator.of(context).pop(),
+          ),
+          style: RoundedButtonStyle.actionSheetLargeRoundedButton().copyWith(
+            height: _Constants.actionHeight,
+            width: _Constants.actionWidth,
+            borderRadius: _Constants.actionCornerRadius,
+          ),
+        ),
+      ),
+    ];
+    listBuilder.add(
+      SizedBox(
+        width: _Constants.actionLineSpace,
+      ),
+    );
+    listBuilder.add(
+      Expanded(
+        child: RoundedButton(
+          viewModel: RoundedButtonViewModel(
+              title: _viewModel.confirmActionText,
+              onTap: () => confirmAndDismiss(context)),
+          style: RoundedButtonStyle.actionSheetLargeRoundedButton().copyWith(
+            height: _Constants.actionHeight,
+            width: _Constants.actionWidth,
+            backgroundColor: _style.actionBackgroundColor,
+            buttonTextStyle: _style.actionTextStyle,
+            borderRadius: _Constants.actionCornerRadius,
+          ),
+        ),
+      ),
+    );
+    return listBuilder;
+  }
+
+  confirmAndDismiss(BuildContext context) {
+    _viewModel.confirmInputAction(_textFieldController.text);
```

Comment:
wonder if we should allow null action here, and if the cancel name not set it becomes a OK dialog message?

Reply:
Cancel is mandatory

Reply:
DONE